src/scd.cpp

```diff
@@ -246,6 +247,25 @@ namespace openre::scd
         return SCD_RESULT_NEXT_TICK;
     }
 
+    // 0x004E43D0
+    static int scd_evt_end(SCE_TASK* sce)
+    {
+        auto eax = sce->Sub_ctr;
```

Comment:
Try to name things as best as you can and avoid the original register names.
In this case I think `sub_ctr` is how many subroutines deep this task is in. So if it is 0, we are at the top and the instruction will terminate the task altogether. Otherwise it just exits from the current subroutine and returns back to the caller.